src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -6,41 +6,51 @@
 # --------------------------------------------------------------------------------------------
 
 # pylint: disable=line-too-long
+from knack.log import get_logger
 from knack.prompting import NoTTYException, prompt_y_n
 from knack.util import CLIError
+from azure.appconfiguration import AzureAppConfigurationClient
+from azure.mgmt.appconfiguration.models import ErrorException
 
 from ._client_factory import cf_configstore
+from ._constants import HttpHeaders
+
+logger = get_logger(__name__)
 
 
 def construct_connection_string(cmd, config_store_name):
     connection_string_template = 'Endpoint={};Id={};Secret={}'
+    # If the logged in user/Service Principal does not have 'Reader' or 'Contributor' role
+    # assigned for the requested AppConfig, resolve_resource_group will raise CLI error
+    resource_group_name, endpoint = resolve_resource_group(cmd, config_store_name)
 
     try:
-        resource_group_name, endpoint = resolve_resource_group(
-            cmd, config_store_name)
         config_store_client = cf_configstore(cmd.cli_ctx)
         access_keys = config_store_client.list_keys(
             resource_group_name, config_store_name)
         for entry in access_keys:
             if not entry.read_only:
                 return connection_string_template.format(endpoint, entry.id, entry.value)
-    except Exception:
+    except Exception as ex:
         raise CLIError(
-            'Cannot find the App Configuration {}. Check if it exists in the subscription that logged in. '.format(config_store_name))
+            'Failed to get access keys for the App Configuration "{}". Make sure that the account that logged in has "Contributor" role assigned for this App Configuration store.\n{}'.format(config_store_name, str(ex)))
```

Comment:
>Contributor [](start = 118, length = 11)

We don't need to specify which role is needed as it can be contributor, reader, owner etc. Maybe just say "Failed to get access keys for the App Configuration "{}". Make sure that the account that logged in has proper role assigned for this App Configuration store.\n{}"